File: frontend/src/api/kbManage.ts
/**
 * 知识库管理 API 封装
 *
 * 全部代理到后端 /api/kb-manage/*，后端再转发到无矩2.0 :8000/api/admin/*
 */
import client from './client'

// ── Types ──────────────────────────────────────────────────────

export interface Collection {
  name: string
  count: number
  type: 'auto_generated' | 'user_imported'
  description: string
}

export interface SyncStatus {
  last_sync: string
  status: string
  fingerprint_methods: number
}

export interface CollectionsResponse {
  collections: Collection[]
  total_docs: number
  sync_status: SyncStatus
}

export interface BrowseDoc {
  id: string
  title?: string
  file_name?: string
  file_format?: string
  preview: string
  preview_length: number
  metadata?: Record<string, any>
  doc_type: string
  imported_at?: string
}

export interface BrowseResponse {
  collection: string
  page: number
  page_size: number
  total: number
  total_pages: number
  docs: BrowseDoc[]
}

export interface DocDetail {
  id: string
  content: string
  metadata?: Record<string, any>
  collection: string
}

export interface SqliteDatabase {
  name: string
  path: string
  description: string
  tables: { name: string; count: number; description: string }[]
}

export interface SqliteTablesResponse {
  databases: SqliteDatabase[]
}

export interface SqliteTableResponse {
  database: string
  table: string
  columns: string[]
  rows: string[][]
  total: number
  page: number
  page_size: number
  total_pages: number
}

export interface ImportTextResponse {
  id: string
  title: string
  collection: string
  chunks: number
  file_name: string | null
  file_format: string | null
}

export interface ImportFileResponse {
  id: string
  title: string
  file_name: string
  file_format: string
  content_length: number
  chunks: number
  collection: string
}

export interface DeleteResponse {
  deleted: boolean
  collection: string
  doc_id: string
}

export type SyncMode = 'backend' | 'frontend' | 'full'

export interface SyncResponse {
  task_id: number
  status: string
  mode: string
  dry_run: boolean
  message: string
}

export interface SyncStatusResponse {
  task_id: number
  status: string
  started_at: string | null
  finished_at: string | null
  result_summary: string | null
  dry_run: boolean
}

export interface SnapshotInfo {
  snapshot_id: string
  created_at: string
  reason: string
  triggered_by: string
  total_chroma_records: number
  sqlite_tables: number
}

export interface SnapshotDetail {
  snapshot_id: string
  created_at: string
  reason: string
  triggered_by: string
  collections: Record<string, number>
  sqlite_tables: { table: string; count: number }[]
}

// ── API Functions ──────────────────────────────────────────────

/** 1. 知识库概览 */
export async function getCollections(): Promise<CollectionsResponse> {
  const res = await client.get('/kb-manage/collections')
  return res.data.data
}

/** 2. 浏览集合内容 */
export async function browseCollection(collection: string, page = 1, pageSize = 20, keyword = ''): Promise<BrowseResponse> {
  const res = await client.get('/kb-manage/browse', { params: { collection, page, page_size: pageSize, keyword } })
  return res.data.data
}

/** 3. 查看单条文档全文 */
export async function getDocument(collection: string, docId: string): Promise<DocDetail> {
  const res = await client.get('/kb-manage/doc', { params: { collection, doc_id: docId } })
  return res.data.data
}

/** 4. SQLite 数据库/表列表 */
export async function getSqliteTables(): Promise<SqliteTablesResponse> {
  const res = await client.get('/kb-manage/sqlite/tables')
  return res.data.data
}

/** 5. SQLite 表内容 */
export async function getSqliteTable(dbName: string, tableName: string, page = 1, pageSize = 20): Promise<SqliteTableResponse> {
  const res = await client.get('/kb-manage/sqlite/table', { params: { db_name: dbName, table_name: tableName, page, page_size: pageSize } })
  return res.data.data
}

/** 6. 导入文本（collection 默认 manual_kb，可指定其他可写集合）*/
export async function importText(
  title: string,
  content: string,
  metadata?: Record<string, any>,
  collection?: string,
): Promise<ImportTextResponse> {
  const res = await client.post('/kb-manage/import', { title, content, metadata, collection })
  return res.data.data
}

/** 7. 导入文件 */
export async function importFile(file: File, title?: string, metadata?: string): Promise<ImportFileResponse> {
  const formData = new FormData()
  formData.append('file', file)
  if (title) formData.append('title', title)
  if (metadata) formData.append('metadata', metadata)
  const res = await client.post('/kb-manage/import/file', formData, {
    headers: { 'Content-Type': 'multipart/form-data' },
  })
  return res.data.data
}

/** 8. 删除文档 */
export async function deleteDocument(collection: string, docId: string): Promise<DeleteResponse> {
  const res = await client.delete('/kb-manage/delete', { data: { collection, doc_id: docId } })
  return res.data.data
}

/** 9. 触发代码同步（T025 3 模式） */
export async function triggerSync(mode: SyncMode = 'backend', dryRun = true): Promise<SyncResponse> {
  // 仅 backend 模式支持 dry_run，frontend/full 直接真实同步
  const query = mode === 'backend' ? `mode=${mode}&dry_run=${dryRun}` : `mode=${mode}`
  const res = await client.post(`/kb-manage/sync?${query}`)
  return res.data.data
}

/** 10. 轮询同步状态 */
export async function getSyncStatus(taskId: number): Promise<SyncStatusResponse> {
  const res = await client.get('/kb-manage/sync/status', { params: { task_id: taskId } })
  return res.data.data
}

// ── 快照与回退（T025 新增）──────────────────────────────────────

/** 11. 创建快照 */
export async function createSnapshot(reason = 'manual'): Promise<{ snapshot_id: string; message: string }> {
  const res = await client.post('/kb-manage/snapshots', null, { params: { reason } })
  return res.data.data
}

/** 12. 快照列表（时间倒序） */
export async function listSnapshots(): Promise<{ snapshots: SnapshotInfo[]; total: number }> {
  const res = await client.get('/kb-manage/snapshots')
  return res.data.data
}

/** 13. 快照详情 */
export async function getSnapshotDetail(snapshotId: string): Promise<SnapshotDetail> {
  const res = await client.get(`/kb-manage/snapshots/${snapshotId}`)
  return res.data.data
}

/** 14. 一键回退（危险操作） */
export async function rollbackSnapshot(snapshotId: string): Promise<{ snapshot_id: string; restored: object }> {
  const res = await client.post('/kb-manage/rollback', null, { params: { snapshot_id: snapshotId } })
  return res.data.data
}

/** 15. 删除快照 */
export async function deleteSnapshot(snapshotId: string): Promise<{ deleted: string }> {
  const res = await client.delete(`/kb-manage/snapshots/${snapshotId}`)
  return res.data.data
}

// ── 图谱查询（深度模式 Agent 2 数据源）────────────────────────

export interface PlatformModule {
  module_name: string
  node_count: number
  description?: string
}

export interface ModulesResponse {
  modules: PlatformModule[]
  total: number
}

export interface ModuleOverview {
  module_name: string
  controllers: GraphNode[]
  apis: GraphNode[]
  frontend_pages: GraphNode[]
  external_deps: GraphNode[]
  services?: GraphNode[]
  description?: string
}

export interface GraphNode {
  node_id?: string
  name?: string
  node_type?: string
  module?: string
  source_file?: string
  [key: string]: any
}

export interface ImpactNode {
  node: GraphNode
  depth: number
  path: string[]
  edge_types_traversed: string[]
}

export interface ImpactResult {
  origin: GraphNode
  direction: 'outgoing' | 'incoming'
  impacted: ImpactNode[]
  summary_by_type: Record<string, number>
  candidates?: GraphNode[]
}

export interface FlowChain {
  [key: string]: any
}

/** 16. 12 业务模块清单（架构快照索引） */
export async function listModules(): Promise<ModulesResponse> {
  const res = await client.get('/kb-manage/modules')
  return res.data.data
}

/** 17. 单模块架构快照（支持中文模块名） */
export async function getModuleOverview(name: string): Promise<ModuleOverview> {
  const res = await client.get(`/kb-manage/modules/${encodeURIComponent(name)}`)
  return res.data.data
}

/** 18. 定向影响范围分析（in=谁依赖我，out=我改波及谁） */
export async function graphImpact(
  node: string,
  direction: 'in' | 'out' = 'in',
  depth = 5,
): Promise<ImpactResult> {
  const res = await client.get('/kb-manage/graph/impact', {
    params: { node, direction, depth },
  })
  return res.data.data
}

/** 19. API 完整调用链（frontend → controller → service → executor） */
export async function graphFlow(api: string): Promise<FlowChain> {
  const res = await client.get('/kb-manage/graph/flow', { params: { api } })
  return res.data.data
}

/** 20. 单个图谱节点详情 */
export async function getGraphNode(nodeId: string): Promise<GraphNode> {
  const res = await client.get(`/kb-manage/graph/node/${encodeURIComponent(nodeId)}`)
  return res.data.data
}

// ── 组件注册表（RenderEngine 消费）─────────────────────

export interface RegistryProp {
  name: string
  type: string
  required: boolean
}

export interface RegistryComponent {
  name: string
  category: 'antd' | 'project' | 'business'
  source_file: string
  props: RegistryProp[]
  subcomponents: string[]
  used_in_pages: string[]
  reuse_count: number
  description?: string
}

// ── 历史 PRD 管理 ─────────────────────────────

export interface PRDInfo {
  prd_id: string
  title: string
  author?: string
  version?: string
  status: 'draft' | 'review' | 'approved' | 'archived'
  related_modules?: string[]
  feature_scope?: string[]
  summary?: string
  created_at: string
  updated_at: string
}

export interface PRDDetail extends PRDInfo {
  content: string
  metadata_json?: Record<string, any>
}

export interface PRDListResponse {
  total: number
  page: number
  page_size: number
  total_pages: number
  items: PRDInfo[]
}

export interface PRDSearchResult {
  prd_id: string
  title: string
  author?: string
  version?: string
  status: string
  summary?: string
  related_modules?: string
  score: number
}

export interface DesignLayoutsResponse {
  total_pages: number
  page_type_counts: Record<string, number>
  most_used_components: string[]
  modules: Record<string, any[]>
}

/** 22. 历史 PRD 列表 */
export async function listPRDs(page = 1, pageSize = 20, keyword = '', status = ''): Promise<PRDListResponse> {
  const res = await client.get('/kb-manage/prds', { params: { page, page_size: pageSize, keyword, status } })
  return res.data.data
}

/** 23. 创建历史 PRD */
export async function createPRD(data: { title: string; content?: string; author?: string; version?: string; status?: string; related_modules?: string[]; feature_scope?: string[]; summary?: string }): Promise<{ prd_id: string; title: string }> {
  const res = await client.post('/kb-manage/prds', data)
  return res.data.data
}

/** 24. PRD 详情（含全文 content） */
export async function getPRDDetail(prdId: string): Promise<PRDDetail> {
  const res = await client.get(`/kb-manage/prds/${prdId}`)
  return res.data.data
}

/** 25. 更新 PRD */
export async function updatePRD(prdId: string, data: Partial<PRDDetail>): Promise<{ prd_id: string; title: string }> {
  const res = await client.put(`/kb-manage/prds/${prdId}`, data)
  return res.data.data
}

/** 26. 删除 PRD */
export async function deletePRD(prdId: string): Promise<{ deleted: string }> {
  const res = await client.delete(`/kb-manage/prds/${prdId}`)
  return res.data.data
}

/** 27. 语义搜索 PRD */
export async function searchPRDs(query: string, topK = 5): Promise<{ query: string; results: PRDSearchResult[] }> {
  const res = await client.post(`/kb-manage/prds/search?query=${encodeURIComponent(query)}&top_k=${topK}`)
  return res.data.data
}

/** 28. 页面布局注册表 */
export async function getDesignLayouts(): Promise<DesignLayoutsResponse> {
  const res = await client.get('/kb-manage/design-layouts')
  return res.data.data
}

/** 21. 获取组件注册表（project/business 组件信息，供 RenderEngine 占位渲染） */
export async function getComponentRegistry(): Promise<RegistryComponent[]> {
  try {
    const res = await client.get('/kb-manage/components/registry')
    return res.data.data || []
  } catch {
    return []
  }
}

File: backend/routers/kb_manage.py
"""
知识库管理 router — 代理到无矩2.0 FastAPI 微服务管理 API

所有接口代理到 http://localhost:8000/api/admin/*，统一响应格式：
    {"status": "success"|"error", "data": {...}, "error": "..."}
"""
import json
import logging
import os
from urllib.parse import quote

import requests
from flask import Blueprint, Response, request, jsonify

logger = logging.getLogger(__name__)

kb_manage_bp = Blueprint("kb_manage", __name__)

# 无矩2.0 管理 API 基地址
ADMIN_BASE = "http://localhost:8000/api/admin"
# 通用请求超时（非 SSE 接口，30s 足够）
_DEFAULT_TIMEOUT = 30


def _proxy_get(path: str, params: dict = None, timeout: int = _DEFAULT_TIMEOUT):
    """通用 GET 代理"""
    try:
        resp = requests.get(f"{ADMIN_BASE}{path}", params=params, timeout=timeout)
        resp.raise_for_status()
        return resp.json()
    except requests.exceptions.ConnectionError:
        return {"status": "error", "data": None, "error": "无法连接到知识库服务（localhost:8000），请确认无矩2.0 已启动。"}
    except requests.exceptions.Timeout:
        return {"status": "error", "data": None, "error": f"知识库服务响应超时（>{timeout}s）。"}
    except requests.exceptions.RequestException as e:
        return {"status": "error", "data": None, "error": f"请求失败: {str(e)}"}


def _proxy_post(path: str, json_body: dict = None, timeout: int = _DEFAULT_TIMEOUT):
    """通用 POST 代理"""
    try:
        resp = requests.post(f"{ADMIN_BASE}{path}", json=json_body, timeout=timeout)
        resp.raise_for_status()
        return resp.json()
    except requests.exceptions.ConnectionError:
        return {"status": "error", "data": None, "error": "无法连接到知识库服务（localhost:8000），请确认无矩2.0 已启动。"}
    except requests.exceptions.Timeout:
        return {"status": "error", "data": None, "error": f"知识库服务响应超时（>{timeout}s）。"}
    except requests.exceptions.RequestException as e:
        return {"status": "error", "data": None, "error": f"请求失败: {str(e)}"}


def _proxy_delete(path: str, json_body: dict = None, timeout: int = _DEFAULT_TIMEOUT):
    """通用 DELETE 代理"""
    try:
        resp = requests.delete(f"{ADMIN_BASE}{path}", json=json_body, timeout=timeout)
        resp.raise_for_status()
        return resp.json()
    except requests.exceptions.ConnectionError:
        return {"status": "error", "data": None, "error": "无法连接到知识库服务（localhost:8000），请确认无矩2.0 已启动。"}
    except requests.exceptions.Timeout:
        return {"status": "error", "data": None, "error": f"知识库服务响应超时（>{timeout}s）。"}
    except requests.exceptions.RequestException as e:
        return {"status": "error", "data": None, "error": f"请求失败: {str(e)}"}


def _proxy_put(path: str, json_body: dict = None, timeout: int = _DEFAULT_TIMEOUT):
    """通用 PUT 代理"""
    try:
        resp = requests.put(f"{ADMIN_BASE}{path}", json=json_body, timeout=timeout)
        resp.raise_for_status()
        return resp.json()
    except requests.exceptions.ConnectionError:
        return {"status": "error", "data": None, "error": "无法连接到知识库服务（localhost:8000），请确认无矩2.0 已启动。"}
    except requests.exceptions.Timeout:
        return {"status": "error", "data": None, "error": f"知识库服务响应超时（>{timeout}s）。"}
    except requests.exceptions.RequestException as e:
        return {"status": "error", "data": None, "error": f"请求失败: {str(e)}"}


def _proxy_files(path: str, files: dict, data: dict, timeout: int = _DEFAULT_TIMEOUT):
    """通用文件上传代理"""
    try:
        resp = requests.post(f"{ADMIN_BASE}{path}", files=files, data=data, timeout=timeout)
        resp.raise_for_status()
        return resp.json()
    except requests.exceptions.ConnectionError:
        return {"status": "error", "data": None, "error": "无法连接到知识库服务（localhost:8000），请确认无矩2.0 已启动。"}
    except requests.exceptions.Timeout:
        return {"status": "error", "data": None, "error": f"知识库服务响应超时（>{timeout}s）。"}
    except requests.exceptions.RequestException as e:
        return {"status": "error", "data": None, "error": f"请求失败: {str(e)}"}


# ── 1. 知识库概览 ──────────────────────────────────────────────


@kb_manage_bp.route("/collections", methods=["GET"])
def get_collections():
    """查看所有 collection 的文档数、类型和同步状态"""
    result = _proxy_get("/collections")
    return jsonify(result)


# ── 2. 浏览集合内容 ─────────────────────────────────────────────


@kb_manage_bp.route("/browse", methods=["GET"])
def browse_collection():
    """分页浏览指定 collection 的文档列表"""
    collection = request.args.get("collection", "")
    if not collection:
        return jsonify({"status": "error", "data": None, "error": "缺少 collection 参数"}), 400

    params = {
        "page": request.args.get("page", 1, type=int),
        "page_size": request.args.get("page_size", 20, type=int),
        "keyword": request.args.get("keyword", ""),
    }
    result = _proxy_get(f"/{collection}/browse", params=params)
    return jsonify(result)


# ── 3. 查看单条文档全文 ─────────────────────────────────────────


@kb_manage_bp.route("/doc", methods=["GET"])
def get_document():
    """查看单条文档全文"""
    collection = request.args.get("collection", "")
    doc_id = request.args.get("doc_id", "")
    if not collection or not doc_id:
        return jsonify({"status": "error", "data": None, "error": "缺少 collection 或 doc_id 参数"}), 400

    result = _proxy_get(f"/{collection}/doc/{doc_id}")
    return jsonify(result)


# ── 4. SQLite 数据库/表结构 ──────────────────────────────────────


@kb_manage_bp.route("/sqlite/tables", methods=["GET"])
def get_sqlite_tables():
    """列出所有 SQLite 数据库和表"""
    result = _proxy_get("/sqlite/tables")
    return jsonify(result)


@kb_manage_bp.route("/sqlite/table", methods=["GET"])
def get_sqlite_table():
    """查看指定 SQLite 表的内容"""
    db_name = request.args.get("db_name", "")
    table_name = request.args.get("table_name", "")
    if not db_name or not table_name:
        return jsonify({"status": "error", "data": None, "error": "缺少 db_name 或 table_name 参数"}), 400

    params = {
        "page": request.args.get("page", 1, type=int),
        "page_size": request.args.get("page_size", 20, type=int),
    }
    result = _proxy_get(f"/sqlite/{db_name}/{table_name}", params=params)
    return jsonify(result)


# ── 6. 导入文档（文本）─────────────────────────────────────────────


@kb_manage_bp.route("/import", methods=["POST"])
def import_text():
    """导入文本内容到知识库

    对齐 T025 文档 4.5：POST /api/admin/import
    请求体可指定 collection（默认 manual_kb），metadata 可含 doc_type 等字段。
    注意：code_frontend 等系统 collection 不可手动导入（由同步流程维护）。
    """
    body = request.get_json(silent=True) or {}
    content = body.get("content", "")
    title = body.get("title", "")
    if not content or not title:
        return jsonify({"status": "error", "data": None, "error": "缺少 content 或 title 参数"}), 400

    payload = {
        "content": content,
        "title": title,
        "collection": body.get("collection", "manual_kb"),
    }
    if body.get("metadata"):
        payload["metadata"] = body["metadata"]

    result = _proxy_post("/import", json_body=payload)
    return jsonify(result)


# ── 7. 导入文档（文件上传）────────────────────────────────────────


@kb_manage_bp.route("/import/file", methods=["POST"])
def import_file():
    """上传文件导入 manual_kb"""
    if "file" not in request.files:
        return jsonify({"status": "error", "data": None, "error": "缺少 file 字段"}), 400

    file = request.files["file"]
    files = {"file": (file.filename, file.stream, file.content_type)}
    data = {}
    if request.form.get("title"):
        data["title"] = request.form["title"]
    if request.form.get("metadata"):
        data["metadata"] = request.form["metadata"]

    result = _proxy_files("/import/file", files=files, data=data)
    return jsonify(result)


# ── 8. 删除文档 ────────────────────────────────────────────────


@kb_manage_bp.route("/delete", methods=["DELETE"])
def delete_document():
    """删除指定文档"""
    body = request.get_json(silent=True) or {}
    collection = body.get("collection", "")
    doc_id = body.get("doc_id", "")
    if not collection or not doc_id:
        return jsonify({"status": "error", "data": None, "error": "缺少 collection 或 doc_id 参数"}), 400

    result = _proxy_delete(f"/{collection}/doc/{doc_id}")
    return jsonify(result)


# ── 9. 触发代码同步（3 模式）──────────────────────────────────────

@kb_manage_bp.route("/sync", methods=["POST"])
def trigger_sync():
    """触发知识库同步（T025 3 模式）

    Query 参数:
        mode (str): 同步模式 backend/frontend/full，默认 backend
        dry_run (bool): true=仅显示 diff 不写入（仅 backend 生效），默认 true

    非 dry_run 模式同步前自动创建快照，可用 GET /api/admin/snapshots 查看。
    """
    mode = request.args.get("mode", "backend")
    dry_run = request.args.get("dry_run", "true")
    # 仅 backend 模式支持 dry_run
    if mode == "backend":
        query = f"/sync?mode={mode}&dry_run={dry_run}"
    else:
        # frontend/full 不走 dry_run，直接真实同步
        query = f"/sync?mode={mode}"
    result = _proxy_post(query)
    return jsonify(result)


# ── 10. 轮询同步状态 ──────────────────────────────────────────────


@kb_manage_bp.route("/sync/status", methods=["GET"])
def get_sync_status():
    """查看同步任务状态

    result_summary 含快照 ID（snapshot: xxx），可用于回退。
    """
    task_id = request.args.get("task_id", "")
    if not task_id:
        return jsonify({"status": "error", "data": None, "error": "缺少 task_id 参数"}), 400

    result = _proxy_get(f"/sync/status/{task_id}")
    return jsonify(result)


# ── 11. KB 快照与回退（T025 新增）──────────────────────────────────


@kb_manage_bp.route("/snapshots", methods=["POST"])
def create_snapshot():
    """手动创建 KB 快照（Chroma 9 collection + SQLite 5 表）

    Query 参数:
        reason (str): 创建原因，默认 manual
    """
    reason = request.args.get("reason", "manual")
    result = _proxy_post(f"/snapshots?reason={reason}")
    return jsonify(result)


@kb_manage_bp.route("/snapshots", methods=["GET"])
def list_snapshots():
    """列出所有快照（按时间倒序）"""
    result = _proxy_get("/snapshots")
    return jsonify(result)


@kb_manage_bp.route("/snapshots/<snapshot_id>", methods=["GET"])
def get_snapshot_detail(snapshot_id: str):
    """查看指定快照详情（含各 collection/表计数）"""
    result = _proxy_get(f"/snapshots/{snapshot_id}")
    return jsonify(result)


@kb_manage_bp.route("/rollback", methods=["POST"])
def rollback_snapshot():
    """一键回退到指定快照（危险操作）

    Query 参数:
        snapshot_id (str): 目标快照 ID
    """
    snapshot_id = request.args.get("snapshot_id", "")
    if not snapshot_id:
        return jsonify({"status": "error", "data": None, "error": "缺少 snapshot_id 参数"}), 400
    result = _proxy_post(f"/rollback?snapshot_id={snapshot_id}")
    return jsonify(result)


@kb_manage_bp.route("/snapshots/<snapshot_id>", methods=["DELETE"])
def delete_snapshot(snapshot_id: str):
    """删除指定快照"""
    result = _proxy_delete(f"/snapshots/{snapshot_id}")
    return jsonify(result)


# ── 12. 图谱查询（深度模式 Agent 2 数据源，KB_DEEP_MODE_PLAN Part A）──


@kb_manage_bp.route("/modules", methods=["GET"])
def list_platform_modules():
    """12 个业务模块清单 + 各模块节点计数（架构快照索引）

    供 PRD 深度模式 Agent 2 匹配功能关键词到平台模块。
    """
    result = _proxy_get("/modules")
    return jsonify(result)


@kb_manage_bp.route("/modules/<module_name>", methods=["GET"])
def get_platform_module(module_name: str):
    """单模块架构快照（controllers / apis / frontend_pages / external_deps）

    路径参数支持中文模块名（如「集群配置管理」），自动 URL 编码。
    """
    encoded = quote(module_name, safe="")
    result = _proxy_get(f"/modules/{encoded}")
    return jsonify(result)


@kb_manage_bp.route("/graph/impact", methods=["GET"])
def graph_impact():
    """定向影响范围分析（KB_DEEP_MODE_PLAN A1）

    Query 参数:
        node (str): 节点名称（模糊匹配，多候选返回全部）或 node_id（精确）
        direction (str): in/incoming=谁依赖我（改我谁受影响）；
                        out/outgoing=我改波及谁。默认 outgoing
        depth (int): BFS 深度上限，默认 5，最大 10
    """
    node = request.args.get("node", "")
    if not node:
        return jsonify({"status": "error", "data": None, "error": "缺少 node 参数"}), 400

    params = {
        "node": node,
        "direction": request.args.get("direction", "outgoing"),
        "depth": request.args.get("depth", 5, type=int),
    }
    result = _proxy_get("/graph/impact", params=params)
    return jsonify(result)


@kb_manage_bp.route("/graph/flow", methods=["GET"])
def graph_flow():
    """API 完整调用链（frontend_service → Controller → Service → Executor）

    Query 参数:
        api (str): API 入口节点名称或 node_id
    """
    api = request.args.get("api", "")
    if not api:
        return jsonify({"status": "error", "data": None, "error": "缺少 api 参数"}), 400

    result = _proxy_get("/graph/flow", params={"api": api})
    return jsonify(result)


@kb_manage_bp.route("/graph/node/<node_id>", methods=["GET"])
def graph_node_detail(node_id: str):
    """单个图谱节点详情（name / type / module / source_file）"""
    encoded = quote(node_id, safe="")
    result = _proxy_get(f"/graph/node/{encoded}")
    return jsonify(result)


# ── 13. 历史 PRD 管理（API_PRD.md）──


@kb_manage_bp.route("/prds", methods=["GET"])
def list_prds():
    """PRD 列表（分页 + 关键词 + 状态筛选）

    Query 参数: page, page_size, keyword, status
    """
    params = {
        "page": request.args.get("page", 1, type=int),
        "page_size": request.args.get("page_size", 20, type=int),
        "keyword": request.args.get("keyword", ""),
        "status": request.args.get("status", ""),
    }
    result = _proxy_get("/prds", params=params)
    return jsonify(result)


@kb_manage_bp.route("/prds", methods=["POST"])
def create_prd():
    """创建历史 PRD（入库 SQLite + 向量化 Chroma）"""
    body = request.get_json(silent=True) or {}
    title = body.get("title", "")
    if not title:
        return jsonify({"status": "error", "data": None, "error": "缺少 title 参数"}), 400
    result = _proxy_post("/prds", json_body=body)
    return jsonify(result)


@kb_manage_bp.route("/prds/<prd_id>", methods=["GET"])
def get_prd_detail(prd_id: str):
    """PRD 详情（含全文 content）"""
    result = _proxy_get(f"/prds/{prd_id}")
    return jsonify(result)


@kb_manage_bp.route("/prds/<prd_id>", methods=["PUT"])
def update_prd(prd_id: str):
    """更新 PRD（SQLite + Chroma 同步更新）"""
    body = request.get_json(silent=True) or {}
    result = _proxy_put(f"/prds/{prd_id}", json_body=body)
    return jsonify(result)


@kb_manage_bp.route("/prds/<prd_id>", methods=["DELETE"])
def delete_prd(prd_id: str):
    """删除 PRD（同步删 SQLite + Chroma 向量）"""
    result = _proxy_delete(f"/prds/{prd_id}")
    return jsonify(result)


@kb_manage_bp.route("/prds/search", methods=["POST"])
def search_prds():
    """语义搜索 PRD（基于 title+summary 向量匹配）"""
    query = request.args.get("query", "")
    top_k = request.args.get("top_k", 5, type=int)
    if not query:
        return jsonify({"status": "error", "data": None, "error": "缺少 query 参数"}), 400
    result = _proxy_post(f"/prds/search?query={quote(query)}&top_k={top_k}")
    return jsonify(result)


@kb_manage_bp.route("/design-layouts", methods=["GET"])
def get_design_layouts():
    """页面布局注册表（22 页，含页面类型/布局组件/浮层标记）"""
    result = _proxy_get("/design-layouts")
    return jsonify(result)


# ── 14. 组件注册表（API_PRD.md，代理微服务）──


@kb_manage_bp.route("/components/registry", methods=["GET"])
def get_component_registry():
    """组件注册表（606 组件，含 props/子组件/使用页面/复用度）

    代理到微服务 /api/admin/component-registry（API_PRD.md）。
    降级：微服务挂 → 503。
    """
    result = _proxy_get("/component-registry")
    return jsonify(result)